content/articles/build-a-contact-form-with-php/index.md

```diff
@@ -0,0 +1,349 @@
+                        <input type="email" name="email" id="email" value="<?php echo htmlspecialchars($email) ?>">
+                        <label for="email">Your email*</label>
+                        <div class="error"><?php echo $errors["email"] ?></div>
+                    </div>
+                    <div class="input-field">
+                        <i class="material-icons prefix">mode_edit</i>
+                        <textarea id="message" class="materialize-textarea" name="message"><?php echo htmlspecialchars($message) ?></textarea>
+                        <label for="message">Your Message*</label>
+                        <div class="error"><?php echo $errors["message"] ?></div>
+                    </div>
+                    <div class="input-field center">
+                        <input type="submit" class="btn-large z-depth-0" name="submit" id="submit" value="Send"></input>
+                    </div>
+
+                </form>
+
+
+            </div>
+        </div>
+
+
+    </main>
+
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/materialize/1.0.0/js/materialize.min.js"></script>
+
+</body>
+
+</html>
+```
+
+**What's happening here**
+
+1. We created a basic form with the **POST** method which ultimately sends the request back to this page.
+
+> To learn more about PHP forms, check out this awesome [article](https://www.section.io/engineering-education/working-with-forms-in-php/).
+
+2. We are also using [materializeCSS](https://materializecss.com/about.html) for some default styling.
+3. And finally we added some PHP variables within our HTML page to echo out either the error messages or the value that you inputted inside your form. Note that we haven't created these variables yet inside our PHP script so it will produce some errors if you refresh your browser.
+
+> The **htmlspecialchars** function is used to turn special characters returned into the ones HTML can understand and properly display. It prevents vulnerabilities like **cross-site scripting** attack (XSS) by replacing all harmful characters with harmless HTML entities. Therefore, always include this function when outputting users input in your HTML.
+
+### Form Validation in PHP
+
+In this section, we will be creating the logic of our web page to validate users input. In the same `index.php`, at the top of the file add the following code:
+
+```php
+<?php
+
+$success = "";
+$error = "";
+$name = $message = $email = "";
+$errors = array('name' => '', 'email' => '', 'message' => '');
+
+?>
+```
+
+**What you should know**
+
+1. We created several variables to store user input and error or success message
+2. We also created an errors array for errors of specific error messages of the input fields.
+
+Navigate back to the homepage and refresh the homepage, all should work well since we have stated all the variables in our PHP script.
+
+> Notice that this PHP script is wrapped inside a `<?php ?>` tag, if not they would not be treated as PHP code.
```

Comment:
```suggestion
Navigate back to the homepage and refresh the homepage, all should work well since we have stated all the variables in our PHP script.

> Notice that this PHP script is wrapped inside a `<?php ?>` tag, if not it would not be treated as PHP code.
```